src/lib/pipeline.ml

```diff
@@ -587,12 +594,57 @@ module Compiler = struct
     in
     compiler.wrap_bam_node pipeline bam_node
 
-  let compile_variant_caller_step ~compiler (pipeline: vcf pipeline) =
+  let rec compile_variant_caller_step ~compiler (pipeline: vcf pipeline) =
     let {processors ; reference_build; work_dir; machine ;} = compiler in
     let result_prefix = work_dir // to_file_prefix pipeline in
     dbg "Result_Prefix: %S" result_prefix;
     let vcf_node =
       match pipeline with
+      | Somatic_variant_caller
+          (som_vc,
+           Bam_pair (
+             Gatk_bqsr (n_bqsr_config, Gatk_indel_realigner (n_gir_conf, n_bam))
+             , 
+             Gatk_bqsr (t_bqsr_config, Gatk_indel_realigner (t_gir_conf, t_bam))
+           )) 
+        when
+          has_option compiler
+            (function `Multi_sample_indel_realignment _ -> true)
+          && n_gir_conf = t_gir_conf ->
+        let normal = compile_aligner_step ~compiler n_bam in
+        let tumor = compile_aligner_step ~compiler t_bam in
+        let bam_list_node =
+          Gatk.indel_realigner
+            ~processors ~reference_build ~run_with:machine ~compress:false
+            ~configuration:n_gir_conf (KEDSL.Bam_workflow_list [normal; tumor])
+        in
+        begin match KEDSL.explode_bam_list_node bam_list_node with
+        | [realigned_normal; realigned_tumor] ->
+          let new_pipeline =
+            Somatic_variant_caller
+              (som_vc,
+               (Bam_pair (
+                   Gatk_bqsr (n_bqsr_config, 
+                              Bam_sample (realigned_normal#product#path,
+                                          realigned_normal)),
+                   Gatk_bqsr (t_bqsr_config, 
+                              Bam_sample (realigned_tumor#product#path,
+                                          realigned_tumor)))))
+          in
+          compile_variant_caller_step ~compiler new_pipeline
+        | other ->
+          failwithf "Gatk.indel_realigner did not return the correct list \
+                     of length 2 (tumor, normal): it gave %d bams"
+            (List.length other)
+        end
+      | Somatic_variant_caller (som_vc, Bam_pair (_, _)) as soma_pipeline
```

Comment:
This is too strong of a condition; the `new_pipeline` created just above already breaks it :)

Fixing this.